core/integration/host_test.go

```diff
@@ -0,0 +1,272 @@
+package core
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.dagger.io/dagger/core"
+	"go.dagger.io/dagger/engine"
+	"go.dagger.io/dagger/internal/testutil"
+)
+
+func TestHostWorkdir(t *testing.T) {
+	t.Parallel()
+
+	var secretRes struct {
+		Host struct {
+			Workdir struct {
+				Read struct {
+					ID core.DirectoryID
+				}
+			}
+		}
+	}
+
+	err := testutil.Query(
+		`{
+			host {
+				workdir {
+					read {
+						id
+					}
+				}
+			}
+		}`, &secretRes, nil)
+	require.NoError(t, err)
+
+	hostRes := secretRes.Host.Workdir.Read.ID
+	require.NotEmpty(t, hostRes)
+
+	var execRes struct {
+		Container struct {
+			From struct {
+				WithMountedDirectory struct {
+					Exec struct {
+						Stdout struct{ Contents string }
+					}
+				}
+			}
+		}
+	}
+
+	err = testutil.Query(
+		`query Test($host: DirectoryID!) {
+			container {
+				from(address: "alpine:3.16.2") {
+					withMountedDirectory(path: "/host", source: $host) {
+						exec(args: ["ls", "/host"]) {
+							stdout { contents }
+						}
+					}
+				}
+			}
+		}`, &execRes, &testutil.QueryOptions{
+			Variables: map[string]interface{}{
+				"host": hostRes,
+			},
+		})
+	require.NoError(t, err)
+
+	// FIXME(vito): this is brittle; it currently finds the README in the root of
```

Comment:
There's some existing tests that create extensions under `testdata` and use a cloak.yaml. You could re-use that pattern here too but just for loading a workdir rather than creating more extensions too. Just requires not using `testutil.Query` and instead starting up the engine directly with explicit workdir+configPath values.

Not a blocker but will need to happen sometime as the way config loading works is going to change in the near future.